### PR TITLE
@tracer.start_as_current_span

### PR DESCRIPTION
**Pull request by tarekziade**
### Closes https://github.com/elastic/connectors-python/issues/961

Adding some timers to track where we spend time

### CODE REVIEW

**Comment by tarekziade** [comment]
nightly e2e test build with the new timers https://buildkite.com/elastic/connectors-python-nightly/builds/674

**Comment by tarekziade** [comment]
sample from the buildkite output:

```
[FMWK][22:05:48][DEBUG] get a doc took 0.0004730224609375 seconds.
[FMWK][22:05:48][DEBUG] _bulk API call _batch_bulk took 0.6930327415466309 seconds.
```

**Comment by tarekziade** [comment]
we are going to get too many logs with one log per doc. Added a `slow_log` option so we just log the ones that are abnormally slow

https://buildkite.com/elastic/connectors-python-nightly/builds/675

File: connectors/es/sink.py [hunk 1]
```diff
@@ -339,6 +340,8 @@ async def get_docs(self, generator):
         Extraction happens in a separate task, when a document contains files.
         """
         logger.info("Starting doc lookups")
+        generator = TracedAsyncGenerator(generator, "get a doc")
```
Comment (vidok) [inline connectors/es/sink.py:343]: Why do you need this generator here?

Comment (tarekziade) [reply connectors/es/sink.py:343]: it's used to wrap the generator that fetches documents (get_docs) so we can trace how long each call to that generator takes

Comment (vidok) [reply connectors/es/sink.py:343]: Can you use the same format you use to decorate `_batch_bulk`?

**Review by vidok — APPROVED** [review]

File: connectors/es/sink.py (hunk 1 above)
Comment (tarekziade) [reply connectors/es/sink.py:343]: Ah good point, I can detect it's an async generator, I have not thought about this, thanks

Comment (tarekziade) [reply connectors/es/sink.py:343]: Done. That was fun!

**Comment by tarekziade** [comment]
oh no, it was already merged. Adding a follow up issue

**Pull request merged**